lib/generic/states.rb

```diff
@@ -285,10 +285,19 @@ def perform_encounter(time, entity, record_encounter = true)
           end
 
           if record_encounter
+            # find closest service provider
+            service = @encounter_class
+            provider = Synthea::Provider.find_closest_service(entity, service)
+            # hash below is added so that procedures during encounters will have a reference to provider
+            entity[:current_provider] = provider
```

Comment:
I know I suggested this approach but upon further consideration I think it should be slightly modified. Nothing in Synthea prevents a person from being in 2 different encounters at once(in 2 different modules), so if that happens then both modules are setting to this same variable and could stomp on each other. The easiest way to prevent that would be making `entity[:current_provider]` a hash, which takes the name of the current module as the key, ex. `entity[:current_provider][@context.name] = provider` . It feels like more of a hack to do it that way, but I think the ideal way would require a lot of changes.